src/EnergyPlus/Construction.cc

```diff
@@ -214,7 +213,8 @@ void ConstructionProps::calculateTransferFunction(EnergyPlusData &state, bool &E
 
         // Obtain thermal properties from the Material derived type
 
-        auto *thisMaterial = state.dataMaterial->Material(CurrentLayer);
+        auto *thisMaterial = dynamic_cast<Material::MaterialChild *>(state.dataMaterial->Material(CurrentLayer));
+        assert(thisMaterial != nullptr);
```

Comment:
Asserting _after_ the dynamic cast looks a little funny, but it's fine.  If the original material layer is null, then this dynamic cast will just point to the same null address without complaint, so it's fine.

My bigger question is about the dynamic_cast at all.  There is certainly a place for it, but I would encourage you to think twice or thrice about using it.  These things are often a sign of some member variables/methods that need to be inherited/overridden.  Anywho, I'll just drop that comment here for now.

Reply:
For nullptr checks before the dynamic_cast, should it be at the `new MaterialBase`?

the nullptr check after the dynamic_cast her is to catch state.dataMaterial->Material(CurrentLayer) being a different type then MaterialChild